torch/_inductor/codegen/cuda/gemm_template.py

```diff
@@ -387,24 +573,32 @@ def define_gemm_instance(
     @staticmethod
     def should_swap_XW(
         bias: IRNode,
-        beta: float,
     ) -> bool:
-        return True
-
-        # TODO(ipiszy): Check whether it's necessary to swap X/W.
-        # strides = bias.get_stride()
-        # if strides[-1] != 1:
-        #     return True
-        # for stride in strides[:-1]:
-        #     if stride != 0:
-        #         return True
-        # return False
+        """
+        Helper method to determine whether we should do an explicit transpose by switching the order of the
+        matmul operands. This might be neccessary when we can't otherwise arrive at the right memory
+        layout for the given Bias operand.
+        """
+        # If bias is row major, swap all M and N dimensions
+        if (
```

Comment:
Note: this method has always been (since original implementation by yingz) a hacky hotfix for flaky CUDA errors that occurred in some CUTLASS GEMM Kernels with linear epilogue. I have reduced it to only those cases where I could actually see these issues in practice ( manual conversion of larger models, too large for unit tests ). As such, it resolves the TODO item that Ying wrote in here. But there's no good theoretical justification I can give for doing it either way, except "This works &  the new version is less hacky and at least as stable as the previous version".

Reply:
Cleaned up the diff a bit to make it easier to review. See also comments above.